x/wasm/internal/keeper/keeper.go

```diff
@@ -93,7 +115,7 @@ func (k Keeper) Create(ctx sdk.Context, creator sdk.AccAddress, wasmCode []byte,
 	}
 	store := ctx.KVStore(k.storeKey)
 	codeID = k.autoIncrementID(ctx, types.KeyLastCodeID)
-	codeInfo := types.NewCodeInfo(codeHash, creator, source, builder)
+	codeInfo := types.NewCodeInfo(codeHash, creator, source, builder, k.getInstantiateAccessConfig(ctx))
```

Comment:
Okay, this is a design question.

Do we allow the contract creator to set who can instantiate? Or do we use a global default?

If we have a global default with "Somebody", this should likley not be a hardcoded "Somebody", but rather the actual creator.

I would propose to add an optional `instantiatePermission` field to the StoreCode method. If not set, then use `k.getInstantiateAccessConfig(ctx)`.

If governance is required to upload code, then it can also check that this `instantiatePermssion` is acceptable (or unset)

Reply:
IMHO it would make sense to let the contract creator decide on the permission setting. It is not required for Gov but can make sense to speed up the next step.
I kept message modifications out of scope in this spike but will push a another commit to be aligned.